osquery/core/init.cpp

```diff
@@ -368,6 +388,54 @@ Initializer::Initializer(int& argc,
   platformSetup();
 }
 
+Initializer::~Initializer() {}
+
+bool terminateActiveOsqueryInstance() {
```

Comment:
Will this function work on windows? I think this would all make the service manager unhappy

I _think_ filesystem permission on the pid file is enough to keep this from being a footgun, especially with the process name check. But it all feels a bit wobbly.

Reply:
I'm not sure if it's possible to add any tests to this

Reply:
The pid of the active osquery process is obtained using `Pidfile::read(FLAGS_pidfile)`, which will open and read the file following these steps:

1. Open `FLAGS_pidfile`
  a. Windows: open or create the file with `FILE_SHARE_READ`, `FILE_SHARE_WRITE`, `FILE_SHARE_DELETE`
  b. Posix: A simple open()
2. Lock the file
  a. Windows: re-open the file using the previous handle, upgrading restrictions: only allow `FILE_SHARE_READ`, enable `FILE_FLAG_DELETE_ON_CLOSE`
  b. Posix: Attempt to lock the file using `flock()`
3. Was the lock obtained?
 a. Yes: also check whether the PID belongs to a process named osqueryd, then kill it
 b. No: ignore the pidfile and keep initializing, this will eventually overwrite it

This was successfully tested on Windows, Linux and macOS

Reply:
So what happens if you have osquery started by the init system, and then you manually run it with `--force`? I assume it kills the daemon version, and starts up. Then the daemon crash loops.